utreexo-validation-bip.md

```diff
@@ -0,0 +1,328 @@
+
+The UTXO proof has 2 elements: the accumulator proof and the leaf data. The
+leaf data provides the necessary UTXO data for block validation that would be
+stored locally for non-Utreexo nodes. The accumulator proof proves that the
+given UTXO hash preimages are committed in the accumulator.
+
+Accumulator proof is defined in BIP-????, and contains two elements:
+
+1. A vector of positions of the UTXO hashes in the accumulator.
+2. A vector of hashes required to hash up to the roots.
+
+For (1), positions are in the order of the leaves that are being proved in
+the accumulator. These are all the inputs in the natural blockchain order that
+excludes the same block spends.
+
+The UTXO hash preimages follow the same ordering as (1) in the accumulator
+proofs. Each of the positions in (1) refer to the UTXO hash preimage in the same
+index.
+
+| Field Name          | Data Type           | Byte Size | Description                             |
+| ------------------- | ------------------- | --------- | --------------------------------------- |
+| Accumulator Proof   | variable byte array | variable  | The Utreexo proof as defined in BIP-????|
+| UTXO hash preimages | variable byte array | variable  | The UTXO data needed to validate all the transaction in the block |
+
+#### UTXO proof validation
+
+For each block, the UTXO proof must be provided with the bitcoin block for
+validation to be possible. Without the UTXO proof, it's not possible to
+validate that the inputs being referenced exists in the UTXO set.
+
+The end result of the UTXO proof validation results us in the vector of UTXO
+hash preimages that are required to perform the rest of the consensus
+validation checks. Note that the resulting data from the UTXO proof validation
+is the same data that would normally be fetched from the locally stored UTXO
+set.
+
+The order of operations for the UTXO proof validation are:
+
+1. Hash the UTXO preimages.
+2. Verify that the UTXO preimages exist in the accumulator with the verification
+   algorithm specified in BIP-????.
+
+### BIP-0030
+
+[`BIP-0030`](https://github.com/bitcoin/bips/blob/master/bip-0030.mediawiki) is an added
+consensus check that prevents duplicate TXIDs. This check and the historical violations
+of this check affect the consensus validation for Utreexo nodes.
+
+### BIP-0030 and BIP-0034 consensus check
+
+Before `BIP-0030`, the Bitcoin consensus rules allowed for duplicate TXIDs. If two
+transactions shared a same TXID, the transaction outputs of the preceding
+transaction would overwrite the previously created UTXOs. It was assumed that
+TXIDs were unique but it's trivially easy to create a transaction that share
+the same `TXID` for coinbase transactions by re-using the same bitcoin address.
+
+`BIP-0030` check is a consensus check that enforces that newly created transactions
+do not have outputs that overwrites an existing UTXO.
+
+`BIP-0034` was a rule where the block height was included in the script signature
```

Comment:
What was called originally "generator transaction" is now more familiarly referred to as a "coinbase transaction" after the "scriptSig" equivalent being called "coinbase field" in that context.

```suggestion
`BIP-0034` introduces a rule that requires the block height to be included in the coinbase field
```